src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetValueTests.cs

```diff
@@ -0,0 +1,374 @@
+            PropertyInfo pi = GetProperty(typeof(MyCoVariantTest), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            var value = pi.GetValue(obj, new Object[] { 1, "2" });
+
+            Assert.Null(value);
+        }
+
+        [Fact]
+        public static void TestGet_Property2()
+        {
+            string propertyName = "Property2";
+            LaterClass obj = new LaterClass();
+            PropertyInfo pi = GetProperty(typeof(LaterClass), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            int value = (int)pi.GetValue(obj);
+
+            Assert.Equal(value, 100);
+        }
+
+        //Try to set indexer Property
+        [Fact]
+        public static void TestSet_Item()
+        {
+            string propertyName = "Item";
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+            string[] h = { "hello" };
+
+            Assert.Equal(pi.Name, propertyName);
+
+            pi.SetValue(obj, "someotherstring", new Object[] { 99, 2, h, "f" });
+
+            Assert.Equal("992f1someotherstring", obj.setValue);
+
+            pi.SetValue(obj, "pw", new Object[] { 99, 2, h, "SOME  string" });
+
+            Assert.Equal("992SOME  string1pw", obj.setValue);
+        }
+
+        //
+        // Negative Tests for PropertyInfo
+        //
+
+        [Theory]
+        [InlineData("PropertyC", typeof(MyCoVariantTest), new Object[] { 1, "2", 3 })]
+        [InlineData("PropertyC", typeof(MyCoVariantTest), null)]
+        public static void TestGet_ThrowsTargetParameterCountException(string propertyName, Type type, Object[]testObj)
+        {
+            Object obj = Activator.CreateInstance(type);
+            PropertyInfo pi = GetProperty(type, propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<TargetParameterCountException>(() =>
+            {
+                var value = pi.GetValue(obj, testObj);
+            });
+        }
+
+        [Theory]
+        [InlineData("Property1", typeof(LaterClass), null)]
+        [InlineData("PropertyC", typeof(MyCoVariantTest), new Object[] { "1", "2" })]
+        public static void TestGet_ThrowsArgumentException(string propertyName, Type type, Object[] testObj)
+        {
+            Object obj = Activator.CreateInstance(type);
+            PropertyInfo pi = GetProperty(type, propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                var value = pi.GetValue(obj, testObj);
+            });
+        }
+
+
+        //Verify PropertyInfo.GetValue throws Exception
+        [Fact]
+        public static void TestGet_ThrowsException()
+        {
+            string propertyName = "PropertyC";
+            Object obj = Activator.CreateInstance(typeof(MyCoVariantTest));
```

Comment:
Nit: As long as we're refactoring these tests, it'd be nice to switch to the guidelines we have for the repo, which state that when there's a C# keyword that maps to a type, prefer the C# keyword, e.g. ```object``` instead of ```Object```, ```string``` instead of ```String```, ```int``` instead of ```Int32```, etc.